### PR TITLE
Extend English ITN rules for cardinal to support hundreds

### PR DESCRIPTION
**Pull request by lociko**
Signed-off-by: Vasyl Spachynskyi <vspachyn@gmail.com>

# What does this PR do ?

Extend ITN rules for English (cardinals) to support hundreds, like: 
- eleven hundred -> 1100
- twenty one hundred -> 2100
- ninety nine hundred and ninety nine -> 9999

**Collection**: [Text Processing]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by ekmb** [comment]
@lociko, thank you for your PR! 

Could you please replace all instances of the cached grammar directory `--cache_dir /home/TestData/nlp/text_norm/ci/grammars/7-29-22'` with `--cache_dir /home/TestData/nlp/text_norm/ci/grammars/10-27-22` in the https://github.com/NVIDIA/NeMo/blob/main/Jenkinsfile to generate new grammars for the CI tests?

**Comment by lociko** [comment]
> @lociko, thank you for your PR!
> 
> Could you please replace all instances of the cached grammar directory `--cache_dir /home/TestData/nlp/text_norm/ci/grammars/7-29-22'` with `--cache_dir /home/TestData/nlp/text_norm/ci/grammars/10-27-22` in the https://github.com/NVIDIA/NeMo/blob/main/Jenkinsfile to generate new grammars for the CI tests?

Done

File: Jenkinsfile [hunk 1]
```diff
@@ -145,23 +145,23 @@ pipeline {
...
           steps {
-            sh 'CUDA_VISIBLE_DEVICES="" python nemo_text_processing/inverse_text_normalization/inverse_normalize.py --language en --text="twenty" --cache_dir /home/TestData/nlp/text_norm/ci/grammars/7-29-22'
+            sh 'CUDA_VISIBLE_DEVICES="" python nemo_text_processing/inverse_text_normalization/inverse_normalize.py --language en --text="twenty" --cache_dir /home/TestData/nlp/text_norm/ci/grammars/10-27-22'
           }
         }
         stage('Test En non-deterministic TN & Run all En TN/ITN tests (restore grammars from cache)') {
           steps {
-            sh 'CUDA_VISIBLE_DEVICES="" python nemo_text_processing/text_normalization/normalize_with_audio.py --text "\$.01" --n_tagged 2 --cache_dir /home/TestData/nlp/text_norm/ci/grammars/7-29-22'
+            sh 'CUDA_VISIBLE_DEVICES="" python nemo_text_processing/text_normalization/normalize_with_audio.py --text "\$.01" --n_tagged 2 --cache_dir /home/TestData/nlp/text_norm/ci/grammars/10-27-22'
             sh 'CUDA_VISIBLE_DEVICES="" pytest tests/nemo_text_processing/en/ -m "not pleasefixme" --cpu --tn_cache_dir /home/TestData/nlp/text_norm/ci/grammars/7-29-22'
```
Comment (ekmb) [inline Jenkinsfile:159]: this one is missing "7-29-22" -> "10-27-22", and it causes the CI test failure.

Comment (lociko) [reply Jenkinsfile]: Thanks!

Comment (lociko) [reply Jenkinsfile]: @ekmb Could you please look at to CI build it fails again. Thanks!

**Comment by ekmb** [comment]
Thank you for your contribution @lociko! The CI was still using the old grammar files for your PR so I added your changes here https://github.com/NVIDIA/NeMo/pull/5435 .

**Pull request closed without merging**